src/code/z_skelanime.c

```diff
@@ -1,176 +1,1675 @@
+
+    skelAnime->limbCount = (u8)limbCount;
+    skelAnime->dListCount = limbHeader->dListCount;
+
+    allocSize = limbCount * sizeof(Vec3s);
+    skelAnime->limbIndex = SEGMENTED_TO_VIRTUAL(limbHeader->limbIndexSeg);
+
+    if ((flags & 8) != 0) {
+        allocSize += 2;
+    }
+
+    if (actorDrawTbl == NULL) {
+        skelAnime->actorDrawTbl = ZeldaArena_MallocDebug(allocSize, "../z_skelanime.c", 2364);
+        skelAnime->unk_24 = ZeldaArena_MallocDebug(allocSize, "../z_skelanime.c", 2365);
+    } else {
+        if (limbBufCount != limbCount) {
+            __assert("joint_buff_num == joint_num", "../z_skelanime.c", 2369);
+        }
+
+        skelAnime->actorDrawTbl = (Vec3s*)ALIGN16((u32)actorDrawTbl);
+        skelAnime->unk_24 = (Vec3s*)ALIGN16((u32)arg6);
+    }
+
+    if ((skelAnime->actorDrawTbl == NULL) || (skelAnime->unk_24 == NULL)) {
+        osSyncPrintf(VT_FGCOL(RED));
+        osSyncPrintf(
+            "Skeleton_Info_Rom_SV_ct メモリアロケーションエラー\n"); // Skeleton_Info_Rom_SV_ct Memory allocation error
+        osSyncPrintf(VT_RST);
+    }
+
+    func_800A3F08(globalCtx, skelAnime, animation, 1.0f, 0.0f, 0.0f, 0, 0.0f);
+}
+
+void func_800A3B8C(SkelAnime* skelAnime) {
+    if (skelAnime->unk_01 < 2) {
+        skelAnime->animUpdate = &func_800A3D70;
+    } else {
+        skelAnime->animUpdate = &func_800A3E0C;
+    }
+    skelAnime->unk_28 = 0.0f;
+}
+
+void func_800A3BC0(GlobalContext* globalCtx, SkelAnime* skelAnime) {
+    skelAnime->animUpdate(globalCtx, skelAnime);
+}
+
+s32 func_800A3BE4(GlobalContext* globalCtx, SkelAnime* skelAnime) {
+    f32 prevUnk28 = skelAnime->unk_28;
+    f32 updateRate = R_UPDATE_RATE * 0.5f;
+
+    skelAnime->unk_28 -= skelAnime->unk_2C * updateRate;
+
+    if (skelAnime->unk_28 <= 0.0f) {
+        func_800A3B8C(skelAnime);
+    }
+
+    func_800A34DC(globalCtx, skelAnime->limbCount, skelAnime->actorDrawTbl, skelAnime->unk_24,
+                  1.0f - (skelAnime->unk_28 / prevUnk28));
+    return 0;
+}
+
+void func_800A3C9C(GlobalContext* globalCtx, SkelAnime* skelAnime) {
+    f32 updateRate;
+
+    func_800A336C(globalCtx, skelAnime->animCurrent, skelAnime->animCurrentFrame, skelAnime->limbCount,
+                  skelAnime->actorDrawTbl);
+    if (skelAnime->unk_28 != 0) {
+        updateRate = R_UPDATE_RATE * 0.5f;
+        skelAnime->unk_28 -= skelAnime->unk_2C * updateRate;
+        if (skelAnime->unk_28 <= 0.0f) {
+            skelAnime->unk_28 = 0.0f;
+            return;
+        }
+        func_800A34DC(globalCtx, skelAnime->limbCount, skelAnime->actorDrawTbl, skelAnime->unk_24, skelAnime->unk_28);
+    }
+}
+
+s32 func_800A3D70(GlobalContext* globalCtx, SkelAnime* skelAnime) {
+    f32 updateRate = R_UPDATE_RATE * 0.5f;
+
+    skelAnime->animCurrentFrame += skelAnime->animPlaybackSpeed * updateRate;
+    if (skelAnime->animCurrentFrame < 0.0f) {
+        skelAnime->animCurrentFrame += skelAnime->unk_14;
+    } else {
+        if (skelAnime->unk_14 <= skelAnime->animCurrentFrame) {
```

Comment:
I think this would be cleaner as an else if, and the same would apply to the function right below.